packages/bitcore-lib-cash/lib/networks.js

```diff
@@ -77,69 +83,73 @@ function prefixToArray(prefix) {
  * @param {Number} data.scripthash - The scripthash prefix
  * @param {Number} data.xpubkey - The extended public key magic
  * @param {Number} data.xprivkey - The extended private key magic
- * @param {Number} data.networkMagic - The network magic number
- * @param {Number} data.port - The network port
- * @param {Array}  data.dnsSeeds - An array of dns seeds
+ * @param {Array}  data.variants - An array of variants
+ * @param {string} data.variants.name - The name of the variant
+ * @param {Number} data.variants.networkMagic - The network magic number
+ * @param {Number} data.variants.port - The network port
+ * @param {Array}  data.variants.dnsSeeds - An array of dns seeds
  * @return Network
  */
 function addNetwork(data) {
-
   var network = new Network();
 
   JSUtil.defineImmutable(network, {
     name: data.name,
     alias: data.alias,
+    is: data.is,
     pubkeyhash: data.pubkeyhash,
     privatekey: data.privatekey,
     scripthash: data.scripthash,
     xpubkey: data.xpubkey,
     xprivkey: data.xprivkey,
   });
 
-  var indexBy = data.indexBy || Object.keys(data);
-
   if (data.prefix) {
-    _.extend(network, {
+    JSUtil.defineImmutable(network, {
       prefix: data.prefix,
       prefixArray: prefixToArray(data.prefix),
     });
   }
 
   if (data.networkMagic) {
-    _.extend(network, {
+    JSUtil.defineImmutable(network, {
       networkMagic: BufferUtil.integerAsBuffer(data.networkMagic)
     });
   }
 
   if (data.port) {
-    _.extend(network, {
+    JSUtil.defineImmutable(network, {
       port: data.port
     });
   }
 
   if (data.dnsSeeds) {
-    _.extend(network, {
+    JSUtil.defineImmutable(network, {
       dnsSeeds: data.dnsSeeds
     });
   }
-  networks.push(network);
-  indexNetworkBy(network, indexBy);
-  return network;
-}
 
-function indexNetworkBy(network, keys) {
-  for(var i = 0; i <  keys.length; i++) {
-    var key = keys[i];
-    var networkValue = network[key];
-    if(!_.isUndefined(networkValue) && !_.isObject(networkValue)) {
-      if(!networkMaps[networkValue]) {
-        networkMaps[networkValue] = [];
+  for (const value of Object.values(network)) {
```

Comment:
What is the point of the indexBy properties on the regtest network?

Reply:
It's to be able to specify which values you want to add to the networkMap. Yeah, it doesn't seem useful to me either. It could be useful in theory. I'm not sure why regtest is is the only one that has it. Maybe a previous regtest had key values that conflicted w/ testnet? :shrug:

Reply:
Or it might have been to make sure that testnet was returned when calling `get` with a key value. The ordering of adding the network should make that unnecessary, though.

I'm fine if you want to remove the indexBy to make the libs more uniform.